dash/include/dash/algorithm/Accumulate.h

```diff
@@ -91,9 +91,9 @@ ValueType accumulate(
   auto l_first     = index_range.begin;
   auto l_last      = index_range.end;
   auto l_result    = std::accumulate(l_first, l_last, init, binary_op);
-  auto result      = 0;
+  ValueType result = 0;
```

Comment:
That should be `ValueType result = result` to support multiplication as `binary_op` and non-basic types.